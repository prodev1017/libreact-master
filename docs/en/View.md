# `<View>`

A shell components that simply renders its children using [`react-universal-interface`](https://github.com/streamich/react-universal-interface).

Useful to temporary store some data in JSX props before the actual render.


## Usage

```jsx
import {View} from 'libreact/lib/View';

<View anything='foo'>
  All children are simply rendered.
</View>
```
