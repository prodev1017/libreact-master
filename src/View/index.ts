import {render} from 'react-universal-interface';

export const View = (props) => render(props, null);
