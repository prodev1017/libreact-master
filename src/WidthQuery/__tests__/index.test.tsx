import {createElement as h} from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {WidthQuery} from '..';
import {View} from '../../View';

describe('<ViewQuery/>', () => {
  it('renders first matching view', () => {
    const jsx = <WidthQuery width={100}>
      <View maxWidth={123}>view 1</View>
      <View maxWidth={123}>view 2</View>
    </WidthQuery>;
    const wrapper = shallow(jsx);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('picks the correct <View/> by maxWidth 1', () => {
    const jsx = <WidthQuery width={700}>
      <View maxWidth={200} />
      <View maxWidth={400}>view 2</View>
      <View maxWidth={600}>view 3</View>
      <View maxWidth={800}>view 4</View>
      <View maxWidth={1000}>view 5</View>
    </WidthQuery>;
    const wrapper = shallow(jsx);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('picks the correct <View/> by maxWidth 2', () => {
    const jsx = <WidthQuery width={700}>
      <View maxWidth={600}>view 2</View>
      <View maxWidth={700}>view 3</View>
      <View maxWidth={800}>view 4</View>
      <View maxWidth={900}>view 5</View>
    </WidthQuery>;
    const wrapper = shallow(jsx);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('treats empty maxWidth as Infinity', () => {
    const jsx = <WidthQuery width={2000}>
      <View maxWidth={600}>view 2</View>
      <View maxWidth={700}>view 3</View>
      <View maxWidth={800}>view 4</View>
      <View>view 5</View>
    </WidthQuery>;
    const wrapper = shallow(jsx);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
