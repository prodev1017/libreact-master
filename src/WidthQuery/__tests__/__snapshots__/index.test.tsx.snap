// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewQuery/> picks the correct <View/> by maxWidth 1 1`] = `
<Component
  maxWidth={800}
>
  view 4
</Component>
`;

exports[`<ViewQuery/> picks the correct <View/> by maxWidth 2 1`] = `
<Component
  maxWidth={800}
>
  view 4
</Component>
`;

exports[`<ViewQuery/> renders first matching view 1`] = `
<Component
  maxWidth={123}
>
  view 1
</Component>
`;

exports[`<ViewQuery/> treats empty maxWidth as Infinity 1`] = `
<Component>
  view 5
</Component>
`;
